### PR TITLE
RESTWS-885: Reset password REST endpoint requires privileges

### PR DESCRIPTION
**Pull request by abertnamanya**
## Description of what I changed

Added filtering of the user by either username or email  from the RestHelperServiceImpl Dao then replaced instead of use of UserService in the PasswordResetController2_2 with Context.getService

## Issue I worked on

see https://issues.openmrs.org/browse/RESTWS-885

## Checklist: I completed these to help reviewers :)

- [ ] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

  No? -> [read here](https://wiki.openmrs.org/display/docs/Pull+Request+Tips) on how to squash multiple commits into one

- [ ] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [ ] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [ ] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [ ] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [ ] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

File: omod-common/src/main/java/org/openmrs/module/webservices/rest/web/api/impl/RestHelperServiceImpl.java [hunk 1]
```diff
@@ -103,15 +111,15 @@ public <T> List<T> getObjectsByFields(Class<? extends T> type, Field... fields)
...
+
 	/**
 	 * @see org.openmrs.module.webservices.rest.web.api.RestHelperService#getPatients(Collection)
 	 */
 	@Override
 	@SuppressWarnings("unchecked")
 	public List<Patient> getPatients(Collection<Integer> patientIds) {
 		List<Patient> ret = new ArrayList<Patient>();
-		
+
```
Comment (sherrif10) [inline omod-common/src/main/java/org/openmrs/module/webservices/rest/web/api/impl/RestHelperServiceImpl.java:122]: @abertnamanya , Am seeing new lines being added ?

File: omod-common/src/main/java/org/openmrs/module/webservices/rest/web/api/impl/RestHelperServiceImpl.java [hunk 2]
```diff
@@ -122,10 +130,30 @@ public List<Patient> getPatients(Collection<Integer> patientIds) {
...
 			}
 		}
-		
+
 		return ret;
 	}
-	
+
+	@Override
+	public User getUserByUsernameOrEmail(String usernameOrEmail) {
```
Comment (sherrif10) [inline omod-common/src/main/java/org/openmrs/module/webservices/rest/web/api/impl/RestHelperServiceImpl.java:138]: Can we also include  unit tests for this method?

Comment (sherrif10) [inline omod-common/src/main/java/org/openmrs/module/webservices/rest/web/api/impl/RestHelperServiceImpl.java:138]: Can we also include  unit tests for this method?

File: omod-2.2/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_2/PasswordResetController2_2.java [hunk 3]
```diff
@@ -11,39 +11,48 @@
...
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RequestMethod;
 import org.springframework.web.bind.annotation.ResponseStatus;
 
+@Component("ResetPasswordComponent")
 @Controller
 @RequestMapping(value = "/rest/" + RestConstants.VERSION_1 + "/passwordreset")
-public class PasswordResetController2_2 extends BaseRestController {
-	
+public class PasswordResetController2_2 extends BaseRestController  implements DaemonTokenAware {
```
Comment (dkayiwa) [inline omod-2.2/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_2/PasswordResetController2_2.java:38]: Why did you make this `DaemonTokenAware`?

Comment (dkayiwa) [reply omod-2.2/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_2/PasswordResetController2_2.java:38]: Here are more examples for you to look at: https://github.com/search?l=&q=org%3Aopenmrs+DaemonTokenAware&ref=advsearch&type=Code

**Pull request closed without merging**